packages/notifications/push/amplify_push_notifications/ios/Classes/PushNotificationEventsStreamHandler.swift

```diff
@@ -0,0 +1,113 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import Foundation
+import Flutter
+
+private let channelNamePrefix = "com.amazonaws.amplify/push_notification/"
+
+enum NativeEvent {
+    case tokenReceived
+    case notificationOpened
+    case foregroundMessageReceived
+
+    var eventName: String {
+        switch self {
+        case .tokenReceived:
+            return "TOKEN_RECEIVED"
+        case .notificationOpened:
+            return "NOTIFICATION_OPENED"
+        case .foregroundMessageReceived:
+            return "FOREGROUND_MESSAGE_RECEIVED"
+        }
+    }
+
+    var eventChannelName: String {
+        return "\(channelNamePrefix)event/\(self.eventName)"
+    }
+}
+
+enum AmplifyPushNotificationsEvent {
+    case payload(payload: [AnyHashable: Any])
+    case error(error: FlutterError)
+}
+
+class PushNotificationEventsStreamHandler: NSObject, FlutterStreamHandler {
+    private let eventChannel: FlutterEventChannel
+    private var eventQueue: [AmplifyPushNotificationsEvent] = []
+    private var eventSink: FlutterEventSink?
+
+    init(eventType: NativeEvent, binaryMessenger: FlutterBinaryMessenger) {
+        eventChannel = FlutterEventChannel(name: eventType.eventChannelName, binaryMessenger: binaryMessenger)
+        super.init()
```

Comment:
Calling `super` in the middle seems weird

```suggestion
        super.init()
        eventChannel = FlutterEventChannel(name: eventType.eventChannelName, binaryMessenger: binaryMessenger)
```

Reply:
This is on purpose, and the compiler enforces this -
1. `eventChannel` needs to be initialized before `super.init()` as it's not a property that exists in the `super`
2. `eventChannel.setStreamHandler(self)` the `self` can only be called after `super.init()`